Modules/_pickle.c

```diff
@@ -360,6 +360,81 @@ _Pickle_FastCall(PyObject *func, PyObject *obj)
 
 /*************************************************************************/
 
+static int
+init_method_ref(PyObject *self, _Py_Identifier *name,
```

Comment:
Add a comment explaining that the purpose is to avoid reference cycles through `self`?